src/messaging/Message.ts

```diff
@@ -115,25 +137,44 @@ export default class Message implements IMessage {
     }
   }
 
+  /**
+   * Verifies that neither the hash of [[Message]] nor the sender's signature on the hash have been tampered with.
+   *
+   * @param encrypted The encrypted [[Message]] object which needs to be decrypted.
+   * @param encrypted.message The encrypted body of the [[Message]] which depends on the [[MessageBodyType]].
+   * @param encrypted.nonce The encryption nonce.
+   * @param encrypted.createdAt The timestamp of the message construction.
+   * @param encrypted.hash The hash of the concatenation of message + nonce + createdAt.
+   * @param encrypted.signature The sender's signature on the hash.
+   * @param senderAddress The sender's public SS58 address of the [[Message]].
+   *
+   */
   public static ensureHashAndSignature(
-    encrypted: IEncryptedMessage,
-    senderAddress: IPublicIdentity['address']
+    { message, nonce, createdAt, hash, signature }: IEncryptedMessage,
```

Comment:
_As above:_ It was not obvious what was actually needed from the message. This way, you still input a message but when looking at the function, you immediately know the required fields from message